components/camel-fhir/src/main/java/org/apache/camel/component/fhir/FhirEndpoint.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.fhir;
+
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.impl.DefaultEndpoint;
+import org.apache.camel.spi.UriEndpoint;
+
+/**
+ * Represents a fhir endpoint.
+ */
+@UriEndpoint(scheme = "fhir", title = "FHIR", syntax = "fhir://", label = "fhir", producerOnly = true)
```

Comment:
This syntax is wrong, you need to use `fhir:xxx` where xxx is required `@UriPath` options

Reply:
The FHIR endpoint doesn't have any path or options. It will only execute the provided [IClientExecutable](https://github.com/jamesagnew/hapi-fhir/blob/0352d487204bf2f5eb0f87e204e77e225d47420b/hapi-fhir-base/src/main/java/ca/uhn/fhir/rest/gclient/IClientExecutable.java#L64) contained in the message body and do some error handling.

@davsclaus The reasoning is that HAPI-FHIR comes with a nice [RestfulClient](https://github.com/jamesagnew/hapi-fhir/blob/0352d487204bf2f5eb0f87e204e77e225d47420b/hapi-fhir-base/src/main/java/ca/uhn/fhir/rest/client/api/IRestfulClient.java#L31) that is instantiated with the server URL and is able to construct every request depending on what FHIR version the server is running. I don't see anything else to add for the moment.

If this feels weird I can remove the Endpoint and have the user execute the request in a processing step.

Let me know what you think